frontend-react/docs/rs-react-networking.md

```diff
@@ -0,0 +1,114 @@
+
+```typescript
+export class New {
+    propertyOne: string = ""
+    propertyTwo: int = -1
+    olds: Old[] = []
+}
+
+class Old {
+    oldPropertyOne: ""
+}
+
+// ... class NewApi is down here
+```
+
+## Implement an API Interface
+
+Now that we have our interface class and data shape classes set up, let's look at how we implement this in a component. React relies on state to hold mutable data that needs to trigger UI updates and effects for async tasks that affect the UI/state. For this reason, our own custom hook, `useNetwork` implements the `useState` and `useEffect` hooks. This reduces the implementation to a single line in a component:
+
+```typescript
+const { loading, data, status, message } = useNetwork<New[]>(NewApi.list())
+```
+
+#### Destructured Response
+
+First, the destructuring is a handy way of accessing all parts of the response to use across your component. The response type follows the following shape: 
+
+```typescript
+export interface ResponseType<T> {
+    loading: boolean;
+    data?: T;
+    status: number;
+    message: string;
+}
+```
+> NOTE: Data is of type T or undefined, so ensure you're checking for data presence through conditional renders and optional chaining patterns.
+
+A departure from the previous library also means parting ways with the previous error boundary pattern, as well. Now, inside of a component (rather than at the level above it), we can check `loading` to conditionally display our `<Spinner/>` and pipe `status` and `message` to any error communication in our user interface with ease.
+
+#### Type Casting
+
+```
+useNetwork<New[]>(...)
+```
+
+Our custom network hook is designed to be generic, meaning we can cast any type in via angle brackets and ensure the response is typed properly. In JavaScript, we wouldn't care, but here, this allows our text editors and IDEs to provide proper auto-complete values, and alert us to type errors, undefined properties, etc.
+
+#### Endpoint Parameter
+
+Finally, the network hook accesses the endpoint parameter that's passed in to make the network call with the proper URL and using the proper AxiosRequestConfig. Because of how this is set up, we are able to reduce the logic within our child class methods to just configuring such an endpoint. Should any further developments be needed, we can expand the `Endpoint` interface and `useNetwork` hook to handle the bulk of the it, rather than having to alter a multitude of network calls in various modules and components.
+
+# Areas to Improve
+
+While this is a great new tool at our disposal, nothing is ever truly perfect. In the future, we should consider the following:
+
+- Needs caching!
```

Comment:
Caching seems pretty important. Does react have a standard cache we can use?

Reply:
Most searches led me down the `useMemo` and `useContext` paths, both of which are good options! Perhaps we use the same method we have that created the `GlobalState` layer but instead of persisting data in browser storage, we keep it isolated to that component and let a refresh of the app clear it out.